### DISCUSSION TITLE
Is dropout necessary?

### DISCUSSION THREAD

**Discussion by adhikarirsr** [pyg-team/pytorch_geometric#2503] [Q&A]
I have noticed that dropout is used quite often in PyG examples. Is dropout used as a regularizer? If so, then why is it favored over other regularizers? Also, doesn't SGD implicitly work as a regularizer?

**Comment by rusty1s** [comment]
This depends. In general, dropout is effective as GNNs tend to heavily overfit and to be more robust to noise in the node features and edges. Batch normalization is effective, too, and there exists other variants of dropout, e.g., edge dropout via `torch_geometric.utils.dropout_adj`.
